engine/src/main/java/org/terasology/rendering/dag/ConditionDependentNode.java

```diff
@@ -16,53 +16,31 @@
 package org.terasology.rendering.dag;
 
 import com.google.common.collect.Lists;
-import org.terasology.context.Context;
-import org.terasology.rendering.world.WorldRenderer;
 
-import java.beans.PropertyChangeEvent;
 import java.beans.PropertyChangeListener;
 import java.util.List;
 import java.util.function.Supplier;
 
 /**
  * TODO: Add javadocs
  */
-public abstract class ConditionDependentNode extends AbstractNode implements PropertyChangeListener {
+public abstract class ConditionDependentNode extends AbstractNode {
     private List<Supplier<Boolean>> conditions = Lists.newArrayList();
 
-    private WorldRenderer worldRenderer;
-
-    protected ConditionDependentNode(Context context) {
-        worldRenderer = context.get(WorldRenderer.class);
-    }
-
     protected void requiresCondition(Supplier<Boolean> condition) {
         conditions.add(condition);
-        checkConditions(); // TODO: better to remove this in near feature
     }
 
     private boolean checkConditions() {
         boolean conditionsAreSatisfied = true;
         for (Supplier<Boolean> condition : conditions) {
             conditionsAreSatisfied = conditionsAreSatisfied && condition.get();
         }
-
-        if (conditionsAreSatisfied != isEnabled()) {
-            setEnabled(conditionsAreSatisfied);
-            // enabling/disabling here means we cannot enable/disable nodes directly:
-            // we must always go through the settings.
-            return true;
-        } else {
-            return false;
-        }
```

Comment:
I don't remember why @tdgunes and I did it this way. I suspect if we did there was a reason, but if you @vampcat think perhaps that reason is no longer there and this code can be simplified so dramatically, why not!

So, how are nodes enabled/disabled now?

Reply:
When generating the task list, it already checks `if (node.isEnabled())`. My tweak just makes `isEnabled()` return `is enabled by user AND has all conditions met` for CDN. So, Nodes are enabled/disabled by the user, and their final status is decided by user-set-state and conditions-being-met.

Reply:
Understood. For the time being it's fine. Please let's keep in mind that in the future we'll probably want disabled nodes and/or nodes whose conditions are not met to be completely out of the graph.   This concept is far from fleshed out but it's good to keep it on the backburner I think.